app/src/lib/dragging/reorderDropzoneManager.ts

```diff
@@ -0,0 +1,127 @@
+import { DraggableCommit } from '$lib/dragging/draggables';
+import type { BranchController } from '$lib/vbranches/branchController';
+import type { Branch, Commit } from '$lib/vbranches/types';
+
+// Exported for type access only
+export class ReorderDropzone {
+	constructor(
+		private branchController: BranchController,
+		private branch: Branch,
+		private entry: Entry
+	) {}
+
+	accepts(data: any) {
+		if (!(data instanceof DraggableCommit)) return false;
+		if (data.branchId !== this.branch.id) return false;
+		if (this.entry.distanceToOtherCommit(data.commit.id) === 0) return false;
+
+		return true;
+	}
+
+	onDrop(data: any) {
+		if (!(data instanceof DraggableCommit)) return;
+		if (data.branchId !== this.branch.id) return;
+
+		const offset = this.entry.distanceToOtherCommit(data.commit.id);
+		this.branchController.reorderCommit(this.branch.id, data.commit.id, offset);
+	}
+}
+
+export class ReorderDropzoneManager {
+	private indexer: Indexer;
+
+	constructor(
+		private branchController: BranchController,
+		private branch: Branch,
+		commits: Commit[]
+	) {
+		this.indexer = new Indexer(commits);
+	}
+
+	get topDropzone() {
+		const entry = this.indexer.get('top');
+
+		return new ReorderDropzone(this.branchController, this.branch, entry);
+	}
+
+	dropzoneBelowCommit(commitId: string) {
+		const entry = this.indexer.get(commitId);
+
+		return new ReorderDropzone(this.branchController, this.branch, entry);
+	}
+}
+
+export class ReorderDropzoneManagerFactory {
+	constructor(private branchController: BranchController) {}
+
+	build(branch: Branch, commits: Commit[]) {
+		return new ReorderDropzoneManager(this.branchController, branch, commits);
+	}
+}
+
+// Private classes used to calculate distances between commtis
+class Indexer {
```

Comment:
I decided to move the indexing and distance calculations into their own classes, because its now on longer required to be publicly exported (like it was when we used the old ReorderDropzone.svelte component)